ckan/lib/search/__init__.py

```diff
@@ -163,44 +163,25 @@ def rebuild(package_id=None, only_missing=False, force=False, refresh=False,
     context = {'model': model, 'ignore_auth': True, 'validate': False,
         'use_cache': False}
 
-    # (canada fork only): background search index rebuilding
-    #TODO: upstream contrib!!
-    task = None
-    if in_background:
-        try:
-            if group_id:
-                _entity_id = group_id
-            else:
-                _entity_id = p.toolkit.config.get('ckan.site_id')
-            task = logic.get_action('task_status_show')(context, {'entity_id': _entity_id,
-                                                                  'task_type': 'search_rebuild',
-                                                                  'key': 'search_rebuild'})
-            task['state'] = 'running'
-            task['last_updated'] = str(datetime.datetime.now(datetime.timezone.utc))
-            logic.get_action('task_status_update')({'session': model.meta.create_local_session(), 'ignore_auth': True}, task)
-        except logic.NotFound:
-            pass
-
     if package_id:
         pkg_dict = logic.get_action('package_show')(context,
             {'id': package_id})
-        log.info('Indexing just package %r...', pkg_dict['name'])
         package_index.remove_dict(pkg_dict)
         package_index.insert_dict(pkg_dict)
+        # (canada fork only): background search index rebuilding
+        #TODO: upstream contrib!!
+        yield package_id, 1, 1, None
     elif package_ids is not None:
-        for package_id in package_ids:
+        # (canada fork only): background search index rebuilding
+        #TODO: upstream contrib!!
+        total_packages = len(package_ids)
+        for counter, package_id in enumerate(package_ids):
```

Comment:
So you don't need to `(counter + 1)` below you can specify the start of the enumeration
```suggestion
        for counter, package_id in enumerate(package_ids, 1):
```